### PR TITLE
ref(docker swarm): remove docker swarm support from API

### LINKED ISSUE #987: Remove support for Docker Swarm
Docker has deprecated Swarm in favor of Docker Swarm Mode, which we already support. We should officially sunset Docker Swarm support in an upcoming release. Existing Docker Swarm users can use previous acs-engine releases if they still need to provision Swarm cluster.

### PR DESCRIPTION
**Pull request by seanknox**
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #987 

**Special notes for your reviewer**:
This PR is intended to formally start deprecation of Docker Swarm as an cluster orchestrator offered by ACS. The Swarm parts and scripts remain; this removes Docker Swarm from the 2017-07-01 and vlabs APIs and from documentation.

**Release note**:

```release-note
Remove Docker Swarm support in future versions of ACS APIs.
```

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1033)

### CODE REVIEW

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by seanknox** [comment]
@anhowe updated on your feedback: the Swarm parts and scripts remain; this removes Docker Swarm from the 2017-07-01 and vlabs APIs and from documentation.

File: test/acse-conf/acse-test.json [hunk 1]
```diff
@@ -9,10 +9,6 @@
       "location": "eastus"
     },
     {
-      "cluster_definition": "swarm.json",
```
Comment (anhowe) [inline test/acse-conf/acse-test.json]: we will need to keep testing swarm, until all headers are gone.  is the "swarm.json" using an older header?

File: test/acse-conf/acse-regression.json [hunk 2]
```diff
@@ -36,16 +36,6 @@
       "location": "eastus"
     },
     {
-      "cluster_definition": "disks-managed/swarm-preAttachedDisks-vmas-windows.json",
```
Comment (anhowe) [inline test/acse-conf/acse-regression.json]: +1

we can remove managed disks for swarm, but we will still need a test to test swarm

File: test/acse-conf/acse-regression.json [hunk 3]
```diff
@@ -71,11 +61,6 @@
       "location": "koreacentral"
     },
     {
-      "cluster_definition": "keyvaultcerts/swarm-windows.json",
```
Comment (anhowe) [inline test/acse-conf/acse-regression.json]: windows is part of old header, so keep the test, just make sure locked to older header.

File: test/acse-conf/acse-pr.json [hunk 4]
```diff
@@ -29,14 +29,6 @@
       "location": "eastus2"
     },
     {
-      "cluster_definition": "disks-managed/swarm-preAttachedDisks-vmas-windows.json",
```
Comment (anhowe) [inline test/acse-conf/acse-pr.json]: managed disks and key vault can be removed for swarm.

But please ensure a test still exists for swarm with the old header.

File: test/acs-engine-test/config/config_test.go [hunk 5]
```diff
@@ -18,10 +18,6 @@ func TestConfigParse(t *testing.T) {
       "skip_validation":false
     },
     {
-      "cluster_definition":"examples/swarm.json",
```
Comment (anhowe) [inline test/acs-engine-test/config/config_test.go]: we'll need to keep this test.

File: pkg/api/v20170701/types.go [hunk 6]
```diff
@@ -313,8 +313,3 @@ func (a *AgentPoolProfile) GetSubnet() string {
 func (a *AgentPoolProfile) SetSubnet(subnet string) {
 	a.subnet = subnet
 }
-
-// IsSwarmMode returns true if this template is for Docker CE orchestrator
-func (o *OrchestratorProfile) IsSwarmMode() bool {
```
Comment (anhowe) [inline pkg/api/v20170701/types.go]: isn't this is still accessed from the template for SwarmMode?

**Comment by seanknox** [comment]
@anhowe Removed from vlabs, but that causes unit tests to fail:
```
=== RUN   TestExpected
--- FAIL: TestExpected (3.94s)
	engine_test.go:32: Loading file testdata/disks-managed/swarm-vmas.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/disks-managed/swarm-vmss.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/disks-managed/swarm-windows-vmas.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/disks-managed/swarm-windows-vmss.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/disks-storageaccount/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/key-vault-certs/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/largeclusters/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/largeclusters/swarmvmas.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/location/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/simple/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/vnet/swarmvnet.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/windows/swarm-vmas.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/windows/swarm.json got error: OrchestratorType has unknown orchestrator: Swarm
	engine_test.go:32: Loading file testdata/windows/swarmWinAndLin.json got error: OrchestratorType has unknown orchestrator: Swarm
```

**Comment by jchauncey** [comment]
@anhowe @seanknox do we want to continue with this PR?

**Comment by seanknox** [comment]
Need buy in from product leadership about a deprecation plan. Closing for now.

**Pull request closed without merging**